src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,149 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
+
+import hudson.Extension;
+
+import hudson.model.UnprotectedRootAction;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
+
+    private GlobalConfig globalConfig;
+
+    private static final Logger LOGGER =
+        Logger.getLogger(WebhookEndpoint.class.getName());
+
+    public WebhookEndpoint() {
+        globalConfig = GlobalConfiguration.all().get(GlobalConfig.class);
+    }
+
+    @Override
+    public String getUrlName() {
+        String url = globalConfig.getSlackOutgoingWebhookURL();
+        if (url == null || url.equals(""))
+            return null;
```

Comment:
I think it needs to return a default value other wise it will cause a NPE here:

https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/model/Jenkins.java#L4295

Issue reported here: https://issues.jenkins-ci.org/browse/JENKINS-33556